src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -82,23 +114,17 @@ export class HuiCardPicker extends LitElement {
     }
 
     return html`
+      <search-input
+        .filter=${this._filter}
+        no-autofocus
+        no-label-float
+        @value-changed=${this._handleSearchChange}
+      ></search-input>
       <div class="cards-container">
-        ${previewCards.map((type: string) => {
-          return html`
-            ${until(
-              this._renderCardElement(type),
-              html`
-                <div class="card spinner">
-                  <paper-spinner active alt="Loading"></paper-spinner>
-                </div>
-              `
-            )}
-          `;
-        })}
-        ${nonPreviewCards.map((type: string) => {
+        ${this._filterCards(this._cards, this._filter).map((card: Card) => {
           return html`
             ${until(
-              this._renderCardElement(type, true),
+              this._renderCardElement(card),
```

Comment:
Can we cache the rendered cards? Now we are looking up entities and sorting the config every time you do a search/type a letter